src/client/providers/queryTransformer.ts

```diff
@@ -0,0 +1,39 @@
+import { Observable, of } from 'rxjs'
+import { flatMap, map, switchMap } from 'rxjs/operators'
+import { FeatureProviderRegistry } from './registry'
+
+export type TransformQuerySignature = (query: string) => Observable<string | null | undefined>
+
+/** Provides transformed queries from the first search extension. */
```

Comment:
Inaccurate now: it is not just the first search extension, it calls all of them.

Also, there is no such concept as a "search extension". An extension can have any number of query transformers. The relevant concept here is "query transformers".

So:

```
/** Transforms search queries using registered query transformers from extensions. */
```